### ISSUE TITLE
When accessing multiple fields of a structured array, existence is not checked and return dtype may be empty

### ISSUE THREAD

**Issue by gerritholl** [numpy/numpy#5632]
When accessing multiple fields of a structured array, numpy fails to verify that all the fields exist.  Instead, it will only return existing fields.  If none exist, it return a strange object that has a dtype that is empty.  Normally, creating an object with an empty dtype is not supposed to be possible (`zeros(shape=(5,), dtype=[])` results as expected in `TypeError: Empty data-type`), so this might actually be a dual bug.

```
In [216]: A = numpy.empty(dtype=[("A", "f4"), ("B", "f8")], shape=(5,))

In [217]: A[["AA"]]
Out[217]: 
array([(), (), (), (), ()], 
      dtype='{'names':[], 'formats':[], 'offsets':[], 'itemsize':12}')

In [232]: A[["AA", "B"]]
Out[232]: 
array([(0.0,), (0.0,), (0.0,), (0.0,), (0.0,)], 
      dtype=[('B', '<f8')])
```

**Comment by ahaldane** [comment]
Duplicate of https://github.com/numpy/numpy/issues/4806
I have a proposed fix in https://github.com/numpy/numpy/pull/5636

**Comment by ahaldane** [comment]
Fixed by #5636
